### PR TITLE
CardEditor timeseries support

### PR DESCRIPTION
**Pull request by JoelArmendariz**
Summary
Adds support for Line Chart configuration in the larger DashboardEditor component.

Change List (commits, features, bugs, etc)
Adds:
`CardEditFormContent`
`CardEditFormSettings`
`DataSeriesFormItem`

Acceptance Test (how to verify the PR)

Go to the `DashboardEditor` `Default` story => add a `Time Series line card` => configure each field in the `CardEditor` and confirm that the JSON payload in the JSON editor modal is updated along with the visible card in the dashboard.
Make sure the JSON updates the editor, and the editor updates the JSON.

### CODE REVIEW

**Review by tay1orjones — CHANGES_REQUESTED** [review]
A couple notes of feedback

<details>
  <summary>1. The sidebar background/layout is broken when the browser window size is not tall enough to accommodate the entire sidebar without a scrollbar.</summary>

  image
</details>


<details>
  <summary>2. I know this isn't directly related to your PR, but I noticed that when the header/title wraps, it doesn't have appropriate padding. @sstone2423 was this accounted for in the original spec? I wonder if it should truncate with an ellipsis instead of wrap?</summary>

  image
</details>

**Comment by JoelArmendariz** [comment]
@tay1orjones Can you provide a screenshot for the layout breaking? I'm having trouble recreating it.

**Review by scottdickerson — APPROVED** [review]
LGTM

File: src/components/CardEditor/CardEditForm/CardEditFormContent.jsx [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+          id="title"
+          labelText={mergedI18n.cardTitle}
+          light
+          onChange={(evt) => onChange({ ...cardJson, title: evt.target.value })}
+          value={title}
+        />
+      </div>
+      <div className={`${baseClassName}--input`}>
+        <TextArea
+          id="description"
```
Comment (stuckless) [inline src/components/CardEditor/CardEditForm/CardEditFormContent.jsx:172]: Should the IDs be merged with the parent id?  ideally IDs should be unique in the DOM but I have  feeling that id="description" would be quite common, should this be something like, {`${this.props.id}_cardeditor_description`}

**Review by stuckless — COMMENTED** [review]
just a comment about the naming of IDs

File: src/components/CardEditor/CardEditForm/CardEditFormContent.jsx (hunk 1 above)
Comment (JoelArmendariz) [reply src/components/CardEditor/CardEditForm/CardEditFormContent.jsx:172]: Good observation, I agree. I will update this.

**Comment by JoelArmendariz** [comment]
@tay1orjones I have addressed the first comment in your feedback.
The second looks like it is an issue with the `<h2>` in the `PageTitleBar` component. I can open a separate issue to address whether to add ellipsis.

**Review by tay1orjones — APPROVED** [review]

**Pull request merged**